### DISCUSSION TITLE
A few misc. questions on Web3Wallet to further my understanding

### DISCUSSION THREAD

**Discussion by ECBSJ** [WalletConnect/walletconnect-monorepo#3783] [Developer Support] [answered]
In regards to Web3Wallet, I have a few questions that I've been unable to wrap my head around:

1. Can there be multiple sessions for 1 pairing? If so, what's the point of that?
2. Why is `acknowledged` always false in the session object?
3. Does `web3wallet.disconnectSession` also disconnect the pairing? Does the pairing need to be disconnected?
4. What is the expiry of a "session_request" event? Meaning when does it expire?

**Comment by ganchoradkov** [comment, marked as answer]
Hey @ECBSJ 
>Can there be multiple sessions for 1 pairing? If so, what's the point of that?

technically yes, but not recommended as it's redudant. We're not blocking muliti sessions but why would you force the user to approve new session when there is one already active?

>Why is acknowledged always false in the session object?

`acknowledged` means the other peer ACKed the session. Immediately after the session is established, it might be `false` (i.e. ACK not yet received) but when both peers are online ( dapp & wallet ), it should be updated to `true`

>Does web3wallet.disconnectSession also disconnect the pairing? Does the pairing need to be disconnected?

Web3Wallet doesn't disconnect a pairing as it could receive new session proposal on that pairing. It saves the user the whole QR scaninig step when pairings are utilized

>What is the expiry of a "session_request" event? Meaning when does it expire?

By default, `session_request` has an expiry of 5 minutes. Dapps could configure the expiry up to 7 days.
